deeppavlov/models/tokenizers/ru_sent_tokenizer.py

```diff
@@ -0,0 +1,200 @@
+    if left_last_word:
+        lw = left_last_word.group(1)
+
+        if lw.lower() in joining_shortenings:
+            return _JOIN
+
+        if _ONLY_RUS_CONSONANTS.search(lw) and lw[-1].islower():
+            return _MAYBE
+
+    pse = _PAIRED_SHORTENING_IN_THE_END.search(left)
+    if pse:
+        s1, s2 = pse.groups()
+        if (s1, s2) in paired_shortenings:
+            return _MAYBE
+
+    right_first_word = _FIRST_WORD.match(right)
+    if right_first_word:
+        rw = right_first_word.group(1)
+        if (lw, rw) in paired_shortenings:
+            return _MAYBE
+
+    if _ENDS_WITH_EMOTION.search(left) and _STARTS_WITH_LOWER.match(right):
+        return _JOIN
+
+    initials = _INITIALS.search(left)
+    if initials:
+        border, _ = initials.groups()
+        if (border or ' ') not in "°'":
+            return _JOIN
+
+    if lw.lower() in shortenings:
+        return _MAYBE
+
+    last_letter = _ENDS_WITH_ONE_LETTER_LAT_AND_DOT.search(left)
+    if last_letter:
+        border, _ = last_letter.groups()
+        if (border or ' ') not in "°'":
+            return _MAYBE
+    if _NUMERATION.match(left):
+        return _JOIN
+    return _SPLIT
+
+
+def ru_sent_tokenize(text: str,
+                     shortenings: Set[str] = SHORTENINGS,
+                     joining_shortenings: Set[str] = JOINING_SHORTENINGS,
+                     paired_shortenings: Set[Tuple[str, str]] = PAIRED_SHORTENINGS) -> List[str]:
+    sentences = []
+    sents = _regex_split_separators(text)
+    si = 0
+    processed_index = 0
+    sent_start = 0
+    while si < len(sents):
+        s = sents[si]
+        span_start = text[processed_index:].index(s) + processed_index
+        span_end = span_start + len(s)
+        processed_index += len(s)
+
+        si += 1
+
+        send = _is_sentence_end(text[sent_start: span_end], text[span_end:],
+                                shortenings, joining_shortenings, paired_shortenings)
+        if send == _JOIN:
+            continue
+
+        if send == _MAYBE:
+            if _STARTS_WITH_LOWER.match(text[span_end:]):
+                continue
+            if _STARTS_WITH_DIGIT.match(text[span_end:]):
+                continue
+
+        if not text[sent_start: span_end].strip():
+            logging.warning("Something went wrong while tokenizing")
+        sentences.append(text[sent_start: span_end].strip())
+        sent_start = span_end
+        processed_index = span_end
+
+    if sent_start != len(text):
+        if text[sent_start:].strip():
+            sentences.append(text[sent_start:].strip())
+    return sentences
+
+
+@register("ru_sent_tokenizer")
+class RuSentTokenizer(Component):
+    """
+    Rule-base sentence tokenizer for Russian language
+    """
+    def __init__(self, shortenings: Set[str] = SHORTENINGS,
+                       joining_shortenings: Set[str] = JOINING_SHORTENINGS,
+                       paired_shortenings: Set[Tuple[str, str]] = PAIRED_SHORTENINGS):
```

Comment:
please, add `*args, **kwargs`. When using components in chainer, they can get additional constructor parameters

Reply:
Done